src/backend/parser/parse_coerce.c

```diff
@@ -787,6 +787,7 @@ coerce_to_domain(Node *arg, Oid baseTypeId, int32 baseTypeMod, Oid typeId,
 		 strcmp(type_name, "nchar") == 0 ||
 		 strcmp(type_name, "varbinary") == 0 ||
 		 strcmp(type_name, "binary") == 0 ||
+		 strcmp(type_name, "sysname") == 0 ||
```

Comment:
This looks incorrect since sysname is not supposed to have typemod.

Reply:
That was one of my trials....removing this line also didn't help.